### DISCUSSION TITLE
API Delete: Entire collection items

### DISCUSSION THREAD

**Discussion by MentalGear** [directus/directus#17060] [Archive]
I started out really liking Directus, but since have become a bit fed up with it.
It started when DirectusCloud support migrated my account from community to standard. Ever since, one of the collections doesn't show any data in the App UI anymore. Although, cloud support has been friendly, it takes always several days to answer and the end result of their evaluation was that: "These entries just don't have data for the fields you are looking at. Possibly a result of the way the data was imported? [note: the problem only happened after support's migration] As this isn't a Directus Cloud issue, I am going to close out this ticket."

Now, it seems my only solution is to delete all items in that collection. (can't delete the collection itself since its linked in numerous places and automatons).

However, easier said than done. Using the flows, I can only delete 100 records at a time (almost 60k records). And when I try to delete items via the API, I also need to indicate item ids. Isn't there a way to drop an **entire** collection ?

```
https://csbackend.directus.app/items/messages/[takes one or more ids]
```
<img width="416" alt="Screenshot 2023-01-09 at 17 33 57" src="https://user-images.githubusercontent.com/2837147/211359068-81bd767d-b9d8-4071-a56b-a6d0e7c52971.png">

**Comment by phazonoverload** [comment]
Hey there,

I won't comment on the issue you present because it has been taken to support who have done an internal investigation and provided the necessary information directly.

However, in response to your question about dropping a collection - you currently cannot do this unless you have a project on our Enterprise Cloud. The Enterprise Cloud provides access to write SQL queries directly, which is how you would achieve this rather than Directus-native functionality.

Otherwise, you can use the API with IDs or or use Flows with a limit as mentioned.

I think this sounds like a solid feature request - I think it would be very useful while actively developing projects to empty out entire collections. A quick glance of the Ideas category in GitHub Discussions doesn't appear to show this already exists, so I would encourage you to open a new discussion there with just the feature request. 

Thanks!
